src/runtime_src/core/common/device.h

```diff
@@ -170,14 +170,23 @@ class device : public ishim
     return qr.get(this, std::forward<Args>(args)...);
   }
 
+  /**
+   * get_section_info() - Get the section info from the input buffer  
+   * received from Driver
+   *
+   */
+  void 
+  get_section_info(const char *buff, axlf_section_kind kind,
+                const char *&sect_info, size_t *sect_size);
+
   /**
    * register_axlf() - Callback from shim after AXLF has been loaded.
    *
    * This function extracts meta data sections as needed.
    */
   XRT_CORE_COMMON_EXPORT
   void
-  register_axlf(const axlf*);
+  register_axlf(const axlf*, const char *info_buff = nullptr);
```

Comment:
I don't think the signature should include info_buff.?

Reply:
Reverted to the original signature.